### PR TITLE
ci: update devcontainer building conditions

### PR DESCRIPTION
**Pull request by cre8ivejp**
I'm configuring the dev container building conditions to build it automatically so we don't forget.

### CODE REVIEW

File: .github/workflows/devcontainer-image-build.yaml [hunk 1]
```diff
@@ -2,6 +2,11 @@ name: Bucketeer Dev Container Build and Push Image
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - ".github/.devcontainer"
```
Comment (cre8ivejp) [inline .github/workflows/devcontainer-image-build.yaml:9]: Any changes in this directory will trigger the workflow when merging to the main branch.

Comment (cre8ivejp) [reply .github/workflows/devcontainer-image-build.yaml:9]: If the other files in this directory don't affect the docker image when they change, we can restrict the trigger to the `Dockerfile` file only.

Comment (Ubisoft-potato) [reply .github/workflows/devcontainer-image-build.yaml:9]: The other files in this directory is the key point to change, `.github/.devcontainer` is fine.

Generally we need to add tools in `devcontainer.json` then build the image.

**Review by Ubisoft-potato — COMMENTED** [review]
👍, LGTM!

**Review by Ubisoft-potato — APPROVED** [review]

**Pull request merged**